File: Level 1/피보나치 수/피보나치 수.cpp
#include<iostream>
using namespace std;
long long fibonacci(int n)
{
	int level = 1;
	long long previous_val = 0;
	long long current_val = 0;
	long long Tmp = 0;
	while (level <= n) {
		if (level == 1) {
			previous_val = 0;
			current_val = 1;
			level++;
		}
		else {
			Tmp = current_val;
			current_val = previous_val + current_val;
			previous_val = Tmp;
			level++;
		}
	}
	return current_val;
}

int main()
{
	int testCase = 10;
	long long testAnswer = fibonacci(testCase);

	cout << testAnswer;
}

File: Level 1/피보나치 수/피보나치 수.py
dp = {}
def fibonacci(num):
    if num == 0:
        return 0
    elif num == 1:
        return 1
    elif num in dp:
        return dp[num]
    else:
        sum = fibonacci(num - 1) + fibonacci(num - 2)
        dp[num] = sum
        return sum
# 아래는 테스트로 출력해 보기 위한 코드입니다.
print(fibonacci(3))